templates/test/comments.html.ep

```diff
@@ -1,69 +1,9 @@
-                        % }
-                        <h4 class="media-heading">
-                            <%= $comment->user->name %> wrote <abbr class="timeago" title="<%= $comment->t_created->datetime() %>Z"><%= format_time($comment->t_created) %></abbr>
-                            % if ($comment->t_created != $comment->t_updated) {
-                                (last edited <abbr class="timeago" title="<%= $comment->t_updated->datetime() %>Z"><%= format_time($comment->t_updated) %></abbr>)
-                            % }
-                        </h4>
-                        <div class="media-comment" id="commentMd_<%= $comment->id %>">
-                            %= $comment->rendered_markdown
-                        </div>
-                        % if (current_user && ((current_user->id eq $comment->user->id))) { # TODO: test also for || current_user->is_admin
-                            <textarea class="form-control comment-editing-control" name="text" rows="5"><%= $comment->text %></textarea>
-                            <button class="btn btn-success btn-circle comment-editing-control" type="submit" name="applyChanges">
-                                <span class="glyphicon glyphicon glyphicon-ok"></span>
-                                Apply changes
-                            </button>
-                            <button class="btn btn-warning btn-circle comment-editing-control" type="reset" name="discardChanges" onclick="hideCommentEditor(<%= $comment->id %>, this.form); return true;">
-                                <span class="glyphicon glyphicon glyphicon-remove"></span>
-                                Discard changes
-                            </button>
-                            <input type="hidden" name="comment_id" value="<%= $comment->id %>">
-                        % }
-                    % end
-                </div>
-            </div>
-        </div>
-    </div>
+    %= include 'comments/comment_row', comment => $comment, edit_action => 'edit_comment', remove_action => 'remove_comment'
 % }
 
 % if (current_user) {
     %= form_for 'add_comment', method => "post", class => "form-horizontal", id => "commentForm", role => "form" => begin
-        <div class="form-group">
-            <label for="text" class="col-sm-1 control-label">Comment
-                <img class="img-circle" src="<%= current_user->gravatar(60) %>">
-            </label>
-            <div class="col-sm-11">
-                <textarea class="form-control" name="text" id="text" rows="5"></textarea>
-            </div>
-        </div>
-        <div class="form-group">
-            <div class="col-sm-offset-1 col-sm-11">
-                <button class="btn btn-success btn-circle" type="submit" id="submitComment">
-                    <span class="glyphicon glyphicon-send"></span>
-                    Submit comment</button>
-            </div>
-        </div>
+        %= include 'comments/add_comment_form_groups'
     % end
 % }
```

Comment:
This file is very small now and as far as I see only included by result.html.ep. Hence it would make sense to embed it directly in results.html.ep.